extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js

```diff
@@ -142,32 +145,39 @@ class LifecycleDeleteObjectTask extends BackbeatTask {
             return process.nextTick(done);
         }
 
-        return this._getMetadata(entry, log, (err, objMD) => {
-            if (err) {
-                return done(err);
-            }
-
-            if (objMD.getLegalHold()) {
-                return done(new ObjectLockedError('object locked'));
-            }
+        if (this.objectMD.getLegalHold()) {
```

Comment:
there is a small difference here (vs before): we _always_ retrieve the metadata, even if the object is not versioned.

This used to be skipped at the beginning of _checkObjectLockState, when the entry has no target version.

What is that specific case: old non-versioned objects from before versioning was enabled?
May this case have a significant impact?

Reply:
The case is: we ignore object lock check when the object is not versioned
Always getting the metadata is something we need to do anyway, as the replication check needs to be done regardless of the versioning state of the object.